### DISCUSSION TITLE
Google SSO does not work properly in a reverse proxy environment

### DISCUSSION THREAD

**Discussion by Taikono-Himazin** [open-webui/open-webui#7588] [Issues]
# Bug Report
## Installation Method

docker
Use nginx as reverse proxy to support SSL

## Environment

- **Open WebUI Version:** v0.4.7
- **Ollama (if applicable):** 0.4.7

- **Operating System:** Win 11
- **Browser (if applicable):** Chrome 131.0.6778.86

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [ ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

I have configured Google SSO.
I can sign in successfully.

## Actual Behavior:

①OpenWebUI login screen
②Click "Sign in with Google" to go to the Google screen
③Authentication with Google is successful, and you are returned to the URL set in GOOGLE_REDIRECT_URI.
④*Bug here* You are redirected to the URL after reverse proxy.
⑤Since you cannot access it from the client PC, an error occurs naturally

URL flow
①
https://[front URL]:3443/auth
②https://accounts.google.com/o/oauth2/v2/auth?response_type=code&client_id=XXXXXXX&redirect_uri=https%3A%2F%2F[front URL]%3A3443%2Foauth%2Fgoogle%2Fcallback&scope=openid+email+profile&state=XXX XXX
③
https://[URL on the front side]:3443/oauth/google/callback?state=XXXXXX&code=XXXXXX&scope=email+profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+openid&authuser=0&hd=XXXX&prompt=none
④*Bug here* https://[URL on the back side]/auth

## Description

I'm running OpenWebUI behind a reverse proxy.

WebUI_URL and GOOGLE_REDIRECT_URI are set correctly as the URL on the front side.

A. The redirect destination in ④ is wrong. It should be set based on WebUI_URL, not fastapi.baseurl.

**Comment by Taikono-Himazin** [comment]
@tjbck 
Why was this moved from an issue to a discussion?

**Comment by edolic** [comment]
We have svoled that by modifying the response.Location header in our Reverse Proxy. Maybe the redirect url for the authentication can be taken from the environment setting not from the request.base_url
